lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -214,6 +229,40 @@ prop_initialBlockHeight s =
   where
     (_, wallet) = initWallet block0 dummyGenesisParameters s
 
+-- Rationale here is that pending transactions contributes towards the b
+prop_countRewardsOnce :: WithPending WalletState -> Property
+prop_countRewardsOnce (WithPending wallet pending rewards)
+    = withMaxSuccess 1000
+    $ counterexample ("Total UTxO:\n" <> pretty' (totalUTxO pending wallet))
+    $ counterexample ("Available UTxO:\n" <> pretty' (availableUTxO pending wallet))
+    $ counterexample ("Pending Transactions:\n" <> pretty' (blockListF pending))
+    $ counterexample ("Pending balance:            " <> show pendingBalance)
+    $ counterexample ("Total Rewards:              " <> show (getCoin rewards))
+    $ counterexample ("Total Balance (w/ pending): " <> show totalWithPending)
+    $ counterexample ("Total Balance (no pending): " <> show totalWithoutPending)
+    $ classify hasPending "has pending"
+    $ classify hasWithdrawals "has withdrawals"
+    $ if hasPending
+      then property (totalWithPending < totalWithoutPending)
+      else property (totalWithPending == totalWithoutPending)
```

Comment:
The property is actually more precise than it says "counting rewards once". It really shows that having pending transactions result in a smaller balance for the wallet, regardless of whether there are withdrawals or not. 

Of course, this only holds in the presence of fees that are non-null.

Reply:
```
only counts rewards once.
      +++ OK, passed 10000 tests:
      45.33% has pending
       5.21% has withdrawals
```

Reply:
👍